styx-flyte-client/src/test/java/com/spotify/styx/flyte/client/TestAdminService.java

```diff
@@ -37,9 +44,70 @@ public void createExecution(final ExecutionOuterClass.ExecutionCreateRequest req
             .newBuilder()
             .setDomain(request.getDomain())
             .setProject(request.getProject())
-            .setName(WF_EXECUTION_ID)
+            .setName(WF_EXECUTION_ID_1)
             .build())
         .build());
     responseObserver.onCompleted();
   }
+
+  @Override
+  public void getExecution(final ExecutionOuterClass.WorkflowExecutionGetRequest request,
+                           final StreamObserver<ExecutionOuterClass.Execution> responseObserver) {
+    responseObserver.onNext(ExecutionOuterClass.Execution
+        .newBuilder()
+        .setId(IdentifierOuterClass.WorkflowExecutionIdentifier
+            .newBuilder()
+            .setProject(request.getId().getProject())
+            .setDomain(request.getId().getDomain())
+            .setName(WF_EXECUTION_ID_1)
+            .build())
+        .build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void terminateExecution(final ExecutionOuterClass.ExecutionTerminateRequest request,
+                                 final StreamObserver<ExecutionOuterClass.ExecutionTerminateResponse> responseObserver) {
+    responseObserver.onNext(ExecutionOuterClass.ExecutionTerminateResponse.getDefaultInstance());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void listExecutions(final Common.ResourceListRequest request,
+                             final StreamObserver<ExecutionOuterClass.ExecutionList> responseObserver) {
+    final var executions = List.of(
+        ExecutionOuterClass.Execution
+            .newBuilder()
+            .setId(IdentifierOuterClass.WorkflowExecutionIdentifier
+                .newBuilder()
+                .setProject(PROJECT)
+                .setDomain(DOMAIN)
+                .setName(WF_EXECUTION_ID_1)
+                .build())
+            .build(),
+        ExecutionOuterClass.Execution
+            .newBuilder()
+            .setId(IdentifierOuterClass.WorkflowExecutionIdentifier
+                .newBuilder()
+                .setProject(PROJECT)
+                .setDomain(DOMAIN)
+                .setName(WF_EXECUTION_ID_2))
+            .build(),
+        ExecutionOuterClass.Execution
+            .newBuilder()
+            .setId(IdentifierOuterClass.WorkflowExecutionIdentifier
+                .newBuilder()
+                .setProject("other_project")
+                .setDomain("other_domain")
+                .setName("other_name"))
+            .build()
+    );
+    responseObserver.onNext(ExecutionOuterClass.ExecutionList.newBuilder()
+        .addAllExecutions(executions.stream()
+            .filter(e -> e.getId().getProject().equals(request.getId().getProject()) && e.getId().getDomain().equals(request.getId().getDomain()))
```

Comment:
I think that this is more complicated that it is needed for the test. You are creating three Executions and here you filter one out.

What about just checking that if project and domain matches the constant you return the list of two else you return an empty list?

Reply:
yeah i think you are right, the filtering is out of scope for this test. I will just refactoring it to simply return the list always.